Jellyfin.Server/Middleware/BaseUrlRedirectionMiddleware.cs

```diff
@@ -45,16 +45,29 @@ public async Task Invoke(HttpContext httpContext, IServerConfigurationManager se
             var localPath = httpContext.Request.Path.ToString();
             var baseUrlPrefix = serverConfigurationManager.GetNetworkConfiguration().BaseUrl;
 
-            if (string.Equals(localPath, baseUrlPrefix + "/", StringComparison.OrdinalIgnoreCase)
-                || string.Equals(localPath, baseUrlPrefix, StringComparison.OrdinalIgnoreCase)
-                || string.Equals(localPath, "/", StringComparison.OrdinalIgnoreCase)
-                || string.IsNullOrEmpty(localPath)
-                || !localPath.StartsWith(baseUrlPrefix, StringComparison.OrdinalIgnoreCase))
+            if (!string.IsNullOrEmpty(baseUrlPrefix))
             {
-                // Always redirect back to the default path if the base prefix is invalid or missing
-                _logger.LogDebug("Normalizing an URL at {LocalPath}", localPath);
-                httpContext.Response.Redirect(baseUrlPrefix + "/" + _configuration[ConfigurationExtensions.DefaultRedirectKey]);
-                return;
+                var startsWithBaseUrl = localPath.StartsWith(baseUrlPrefix, StringComparison.OrdinalIgnoreCase);
+
+                if (!startsWithBaseUrl
+                    && localPath.EndsWith("/health", StringComparison.OrdinalIgnoreCase))
+                {
+                    _logger.LogDebug("Redirecting /health check");
+                    httpContext.Response.Redirect(baseUrlPrefix + "/health");
+                    return;
+                }
+
+                if (string.Equals(localPath, baseUrlPrefix + "/", StringComparison.OrdinalIgnoreCase)
+                    || string.Equals(localPath, baseUrlPrefix, StringComparison.OrdinalIgnoreCase)
+                    || string.Equals(localPath, "/", StringComparison.OrdinalIgnoreCase)
+                    || string.IsNullOrEmpty(localPath)
+                    || !startsWithBaseUrl)
```

Comment:
The order and expressions could be improved wrt. fast path, allocations etc. I'm also wondering why localPath == '/' is checked?
```suggestion
                if (!startsWithBaseUrl
                    || string.IsNullOrEmpty(localPath)
                    || localPath.Length == baseUrlPrefix.Length
                    || (localPath.Length == baseUrlPrefix.Length + 1 && localPath.EndsWith('/')))
```

Reply:
I've simplified it further and fixed an issue where requesting `/health/` wouldn't properly redirect. Everything still redirects properly for me with this change